command/command.go

```diff
@@ -518,23 +518,38 @@ func runCheckersLoop(app *App, termCheckerCh <-chan struct{}, quit <-chan struct
 			logger.Debugf("RunChekcerLoop: Extend the delay to %d seconds. There are %d reports.", reportCheckDelay, len(reports))
 		}
 
-		partialReports := make([]*checks.Report, 0, checkReportMaxSize)
-		for i, report := range reports {
-			logger.Debugf("reports[%d]: %#v", i, report)
-			partialReports = append(partialReports, report)
-			if len(partialReports) >= checkReportMaxSize {
-				reportCheckMonitors(app, partialReports)
-				partialReports = make([]*checks.Report, 0, checkReportMaxSize)
+		// "" means no CustomIdentifier, which means the host running this agent itself.
+		reportsByCustomIdentifier := map[string][]*checks.Report{}
+		for _, report := range reports {
+			customIdentifier := ""
+			if report.CustomIdentfier != nil {
+				customIdentifier = *report.CustomIdentfier
+			}
+			if _, exists := reportsByCustomIdentifier[customIdentifier]; !exists {
+				reportsByCustomIdentifier[customIdentifier] = make([]*checks.Report, 0, checkReportMaxSize)
+			}
+			reportsByCustomIdentifier[customIdentifier] = append(reportsByCustomIdentifier[customIdentifier], report)
+			if len(reportsByCustomIdentifier[customIdentifier]) >= checkReportMaxSize {
+				reportCheckMonitors(app, customIdentifier, reportsByCustomIdentifier[customIdentifier])
+				delete(reportsByCustomIdentifier, customIdentifier)
 				time.Sleep(time.Duration(reportCheckDelay) * time.Second)
 			}
 		}
-		reportCheckMonitors(app, partialReports)
+		for customIdentifier, partialReports := range reportsByCustomIdentifier {
+			reportCheckMonitors(app, customIdentifier, partialReports)
+		}
 	}
 }
 
-func reportCheckMonitors(app *App, reports []*checks.Report) {
+func reportCheckMonitors(app *App, customIdentifier string, reports []*checks.Report) {
+	host := app.Host
+	if customIdentifier != "" {
+		if h, ok := app.CustomIdentifierHosts[customIdentifier]; ok {
+			host = h
+		}
```

Comment:
When host of custom identifier is not found, the report is posted to app.Host but it is inconsistent with metric plugins. See https://github.com/mackerelio/mackerel-agent/blob/5d4632d/command/command.go#L379-L385. Is it ok?

Reply:
Not ok!!!!!!! Thank you for review.
The behavior should be same as metric plugins.